control/nlsys.py

```diff
@@ -1663,87 +1671,190 @@ def ivp_rhs(t, x):
         success=soln.success, message=message)
 
 
-def find_eqpt(sys, x0, u0=None, y0=None, t=0, params=None,
-              iu=None, iy=None, ix=None, idx=None, dx0=None,
-              return_y=False, return_result=False):
-    """Find the equilibrium point for an input/output system.
+class OperatingPoint(object):
+    """A class for representing the operating point of a nonlinear I/O system.
+
+    The ``OperatingPoint`` class stores the operating point of a nonlinear
+    system, consisting of the state and input vectors for the system.  The
+    main use for this class is as the return object for the
+    :func:`find_operating_point` function and as an input to the
+    :func:`linearize` function.
+
+    Attributes
+    ----------
+    states : array
+        State vector at the operating point.
+    inputs : array
+        Input vector at the operating point.
+    outputs : array, optional
+        Output vector at the operating point.
+    result : :class:`scipy.optimize.OptimizeResult`, optional
+        Result from the :func:`scipy.optimize.root` function, if available.
+
+    """
+    def __init__(
+            self, states, inputs=None, outputs=None, result=None,
+            return_outputs=False, return_result=False):
+        self.states = states
+        self.inputs = inputs
+
+        if outputs is None and return_outputs and not return_result:
+            raise SystemError("return_outputs specified by no y0 value")
```

Comment:
```suggestion
            raise SystemError("return_outputs specified but no y0 value")
```

Reply:
Here and below are both non-standard uses of [SystemError, which normally is for "when the [Python] interpreter finds an internal error"](https://docs.python.org/3/library/exceptions.html#SystemError). I recommend [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) instead, which is also used elsewhere in this module for inconsistent arguments.